### PR TITLE
Fix file.managed on Windows with test=True

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fix file.managed with makedirs=True on Windows when the target directory structure is missing and test=True

Regression caused by https://github.com/saltstack/salt/pull/39773

Fix formatting issues in the docs

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/43328

### Tests written?
No

### CODE REVIEW

**Comment by rallytime** [comment]
@twangboy It would probably be good to add a quick unit test for this so the regression doesn't happen again.

**Comment by twangboy** [comment]
@rallytime Added some unit tests...

File: salt/modules/win_file.py [hunk 1]
```diff
@@ -818,8 +818,10 @@ def stats(path, hash_type='sha256', follow_symlinks=True):
 
         salt '*' file.stats /etc/passwd
     '''
+    # This is to mirror the behavior of file.py. `check_file_meta` expects an
+    # empty dictionary when the file does not exist
```
Comment (cachedout) [inline salt/modules/win_file.py]: I tend to wonder if this should be changed in the other direction. Wouldn't it be cleaner to raise a specific error and then check for it and display it to the user, versus just an empty dict?

Comment (twangboy) [reply salt/modules/win_file.py]: @cachedout Changed this back, added try/except block to calling function

**Comment by cachedout** [comment]
It would appear that all the file tests are now broken.

**Comment by twangboy** [comment]
Damn....

**Comment by twangboy** [comment]
@rallytime I don't know why the GPG test is failing. It looks to me like all the commits are signed...

**Comment by cachedout** [comment]
Given that this is a sensitive area and that it's a regression fix, I would like to see additional tests added here to guide future development before we review this.

**Comment by twangboy** [comment]
@cachedout There are unit tests... what additional tests do you think we should add here?

**Comment by twangboy** [comment]
Rebased...

**Comment by cachedout** [comment]
Looked at tests in place. They look sufficient.

**Comment by rallytime** [comment]
@twangboy There are two related tests here failing on the python 3 side. Can you take a look?

**Comment by twangboy** [comment]
Rebased...

**Comment by rallytime** [comment]
The Python 3 tests here are still failing. They're related to this change. @twangboy Can you take another look? :)

**Comment by cachedout** [comment]
Bump @twangboy

**Comment by twangboy** [comment]
Rebased...

**Comment by rallytime** [comment]
re-run py2

**Comment by twangboy** [comment]
Rebased...

**Comment by twangboy** [comment]
Rebased...

**Comment by twangboy** [comment]
Rebased...

**Review by terminalmage — APPROVED** [review]

**Pull request merged**